File: id1/Detailed/id1_detailed.go
package main

import(
	"fmt"
)

func sum(values []int) int{
	var sum int
	sum = 0

	for i:=0; i < len(values); i++{
		sum+=values[i]
	}

	return sum
}

func main(){
	var multiples_of_3 []int
	var multiples_of_5 []int
	var multiples_of_3_and_5 []int
	const limit=1000

	for i:=1; i < limit; i++ {
		if( i%3 == 0){
			multiples_of_3 = append(multiples_of_3,i)
		}
		if(i%5 == 0){
			multiples_of_5 = append(multiples_of_5,i)
		}
		if( (i%3==0) || (i%5 == 0)){
			multiples_of_3_and_5 = append(multiples_of_3_and_5,i)
		}
	}
	
//	fmt.Println("List of multiples of 3\n",multiples_of_3,"\n")
//	fmt.Println("List of multiples of 5\n",multiples_of_5,"\n")
//	fmt.Println("List of multiples of 3 or 5\n",multiples_of_3_and_5,"\n")

	fmt.Println("The sum of multiples of 3 is",sum(multiples_of_3))
	fmt.Println("The sum of multiples of 5 is",sum(multiples_of_5))
	fmt.Println("The sum of multiples of 3 or 5 is",sum(multiples_of_3_and_5))
}

File: id1/Detailed/id1_detailed.py
#!/usr/bin/python3


def sum_total(values):
    sum=0
    for i in range(0,len(values)):
        sum+=values[i]
    return sum


limit=1000

multiples_of_3 = [0]
multiples_of_5 = [0]
multiples_of_3_and_5 = [0]

for i in range(0,limit):
    if(i%3==0):
        multiples_of_3.append(i)
    if(i%5==0):
        multiples_of_5.append(i)
    if(i%3==0 or i%5==0):
        multiples_of_3_and_5.append(i)

print("The total sum of multiples of 3 is ",sum_total(multiples_of_3))
print("The total sum of multiples of 5 is ",sum_total(multiples_of_5))
print("The total sum of multiples of 3 or 5 is ",sum_total(multiples_of_3_and_5))
